packages/next/src/build/webpack/loaders/next-app-loader.ts

```diff
@@ -543,7 +543,9 @@ const nextAppLoader: AppLoader = async function nextAppLoader() {
     middlewareConfig,
   }
 
-  const extensions = pageExtensions.map((extension) => `.${extension}`)
+  const normalizedPageExtensions = typeof pageExtensions === "string" ? [pageExtensions] : pageExtensions
```

Comment:
`pageExtensions` should always be an array of strings, can you add a test case where this is slipping through?

x-ref: https://github.com/vercel/next.js/blob/1df2f3ef3c5079eaeb3a4e84df088e8c0697ddb3/packages/next/src/server/config-shared.ts#L617

Reply:
https://github.com/vercel/next.js/blob/7a7a25ae293851e40ca19e4ed7194bbc63c47f1a/packages/next/src/build/webpack/loaders/next-app-loader.ts#L516

seems like this line is giving string when there is a single element in `next.config.js`, but i could not find its definition. could you please help?

Reply:
Ah yea that must be from the loader options stringifying/parsing since it's encoded to a querystring. This fix seems good considering that we'll just want a regression test in one of our test suites to catch this

Reply:
i found a test for babel loader but not for webpack. where are we currently writing regression tests for `next-app-loader`?

Reply:
Looks like for next internal assets still needs js extension for some nextjs extensions

```
Module not found: Can't resolve 'next/dist/compiled/react-refresh/runtime'
```